src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java

```diff
@@ -781,6 +810,7 @@ public int hashCode() {
         result = 31 * result + (dockerCommand != null ? dockerCommand.hashCode() : 0);
         result = 31 * result + (hostname != null ? hostname.hashCode() : 0);
         result = 31 * result + (user != null ? user.hashCode() : 0);
+        result = 31 * result + (extraGroups != null ? extraGroups.hashCode() : 0);
```

Comment:
This isn't a comment about _this_ line but github doesn't let me comment on lines that haven't been changed and this is the closest we get...

There's an inner class called `DescriptorImpl` and that needs to have a `public FormValidation doCheckExtraGroupsString(@QueryParameter String extraGroupsString)` method added to it.
The purpose of that code is to check that we can turn the user's text into something we can persist and use, and let the user know if they've typed rubbish into the field.
The method `doCheckVolumesFromString` might be a good one to use as a basis - it does a "can we parse this? if not, catch the exception and let the user know" test.